src/app/components/grid-list/grid-list.component.ts

```diff
@@ -25,15 +25,25 @@ export class GridListComponent {
     const navigationExtras: NavigationExtras = {
       queryParams: {
         name: item.name,
-        navigatedFrom: this.route,
+        type: this.route,
       },
     };
 
-    if (this.route === 'categoryPlaylists') {
-      return this.router.navigate([`/tracks`, item.id], navigationExtras);
+    switch (item.type) {
+      case 'album':
+        navigationExtras.queryParams.type = 'album';
+        break;
+      case 'artist':
+        navigationExtras.queryParams.type = 'artist';
+        break;
+      case 'playlist':
+        navigationExtras.queryParams.type = 'playlist';
```

Comment:
What's the reason of this switch case statement?

Reply:
In track-list component I use navigationExtras.queryParams.type to define which call should I execute (to search for artists or playlists or albums)

Reply:
'navigationExtras.queryParams.type = item.type' - isn't that what you need?